src/test/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsServiceTest.kt

```diff
@@ -286,6 +339,50 @@ class IncentiveReviewsServiceTest {
     )
   }
 
+  @Test
+  fun `throw exception if cannot calculate next review date for one bookingId`(): Unit = runBlocking {
+    // Given - we only have prisonerIepLevel records for 110001
+    whenever(prisonApiService.getLocation(any())).thenReturnLocation("MDI-2-1")
+    whenever(offenderSearchService.findOffenders(any(), any(), any(), any()))
+      .thenReturnOffenders(
+        listOf(
+          offenderSearchPrisoner("A1409AE", 110001),
+          offenderSearchPrisoner("G6123VU", 110002),
+        )
+      )
+
+    whenever(prisonerIepLevelRepository.findAllByBookingIdInAndCurrentIsTrueOrderByReviewTimeDesc(eq(listOf(110001, 110002))))
+      .thenReturn(flowOf(prisonerIepLevel(110001)))
+
+    // When
+    assertThatThrownBy {
+      runBlocking { incentiveReviewsService.reviews("MDI", "MDI-2-1", "STD") }
+    }.isInstanceOf(ListOfDataNotFoundException::class.java)
+      .hasMessage("No incentive levels found for ID(s) [110002]")
+  }
+
+  @Test
+  fun `throw exception if cannot calculate next review date for all bookingIds`(): Unit = runBlocking {
```

Comment:
Same observation made [in this comment](https://github.com/ministryofjustice/hmpps-incentives-api/pull/257/files#r1024217594) for this test :)